### PR TITLE
Changes for RHDEVDOCS-5250: Doc; Configure pipelines-as-code on Git import (addition)

### PR DESCRIPTION
**Pull request by shreyasiddhartha**
Change type: Doc update; Doc: Configure pipelines-as-code on Git import (addition)
Doc JIRA: https://issues.redhat.com/browse/RHDEVDOCS-5250 
Dev JIRAs: https://issues.redhat.com/browse/ODC-7128
Fix Version: Openshift 4.13 Freeze

Doc Preview: https://59562--docspreview.netlify.app/openshift-enterprise/latest/applications/creating_applications/odc-creating-applications-using-developer-perspective.html#odc-importing-codebase-from-git-to-create-application_odc-creating-applications-using-developer-perspective

SME Review: @vikram-raj 
QE Review: @sanketpathak 
Peer Review: @rolfedh

### CODE REVIEW

**Comment by sanketpathak** [comment]
LGTM

**Review by vikram-raj — COMMENTED** [review]
LGTM

File: modules/odc-importing-codebase-from-git-to-create-application.adoc [hunk 1]
```diff
@@ -55,7 +55,18 @@ The resource name must be unique in a namespace. Modify the resource name if you
...
 ====
 
-. In the *Pipelines* section, select *Add Pipeline*, and then click *Show Pipeline Visualization* to see the pipeline for the application. A default pipeline is selected, but you can choose the pipeline you want from the list of available pipelines for the application. If *Configure PAC* is checked and the GitHub App is set up, you can see *Use GitHub App* and *Setup a webhook* options. If GitHub App is not set up, you can see *Setup a webhook* option.
+. In the *Pipelines* section, select *Add Pipeline*, and then click *Show Pipeline Visualization* to see the pipeline for the application. A default pipeline is selected, but you can choose the pipeline you want from the list of available pipelines for the application.
+
++
+If *Configure PAC* is checked and the GitHub App is set up, you can see *Use GitHub App* and *Setup a webhook* options. If GitHub App is not set up, you can only see *Setup a webhook* option. Use *Setup a webhook* option to add a webhook to your repository. On your GitHub repository or organization, perform the following steps to add a webhook to your repository:
```
Comment (rolfedh) [inline modules/odc-importing-codebase-from-git-to-create-application.adoc:61]: Maybe something like:
```suggestion
. Add a webhook to your repository. If *Configure PAC* is checked and the GitHub App is set up, you can see the *Use GitHub App* and *Setup a webhook* options. If GitHub App is not set up, you can only see the *Setup a webhook* option. In that case, use the *Setup a webhook* option to add a webhook to your repository.
```

File: modules/odc-importing-codebase-from-git-to-create-application.adoc [hunk 2]
```diff
@@ -55,7 +55,18 @@ The resource name must be unique in a namespace. Modify the resource name if you
...
+
+.. Got to *Settings* -> *Webhooks* and click *Add webhook*.
```
Comment (rolfedh) [inline modules/odc-importing-codebase-from-git-to-create-application.adoc:63]:
```suggestion
.. Go to *Settings* -> *Webhooks* and click *Add webhook*.
```

**Review by rolfedh — COMMENTED** [review]
Offered some suggestions...

File: modules/odc-importing-codebase-from-git-to-create-application.adoc (hunk 1 above)
Comment (shreyasiddhartha) [reply modules/odc-importing-codebase-from-git-to-create-application.adoc:61]: The Pipelines section is already a step _8_ of the Import from Git workflow that is dedicated only to pipelines. I'm not sure if we should add this as another step. I've made some changes to the content and have not deleted the line _On your GitHub repository or organisation, perform the following steps to add a webhook to your repository:_. This was requested by the SME's to specifically mention that a developer should add the webhook manually. Do let me know if this is okay! Thanks

File: modules/odc-importing-codebase-from-git-to-create-application.adoc [hunk 3]
```diff
@@ -55,7 +55,18 @@ The resource name must be unique in a namespace. Modify the resource name if you
 To set the default resource preference for importing an application, go to *User Preferences* -> *Applications* -> *Resource type* field. The *Serverless Deployment* option is displayed in the *Import from Git* form only if the {ServerlessOperatorName} is installed in your cluster. The *Resources* section is not available while creating a serverless function. For further details, refer to the {ServerlessProductName} documentation.
 ====
 
-. In the *Pipelines* section, select *Add Pipeline*, and then click *Show Pipeline Visualization* to see the pipeline for the application. A default pipeline is selected, but you can choose the pipeline you want from the list of available pipelines for the application. If *Configure PAC* is checked and the GitHub App is set up, you can see *Use GitHub App* and *Setup a webhook* options. If GitHub App is not set up, you can see *Setup a webhook* option.
+. In the *Pipelines* section, select *Add Pipeline*, and then click *Show Pipeline Visualization* to see the pipeline for the application. A default pipeline is selected, but you can choose the pipeline you want from the list of available pipelines for the application.
+
++
```
[the remark this answers is not in the document]
Comment (shreyasiddhartha) [reply modules/odc-importing-codebase-from-git-to-create-application.adoc:60]: I have removed the empty space from line 59. I'm not sure that line 61 should be a step as it's not a procedure, but specifying a Pipeline related update in the Git Import workflow.

File: modules/odc-importing-codebase-from-git-to-create-application.adoc [hunk 4]
```diff
@@ -64,9 +64,18 @@ The *Add pipeline* checkbox is checked and *Configure PAC* is selected by defaul
...
-
 ====
 
+
+. Add a webhook to your repository. If *Configure PAC* is checked and the GitHub App is set up, you can see the *Use GitHub App* and *Setup a webhook* options. If GitHub App is not set up, you can only see the *Setup a webhook* option:
+
+.. Go to *Settings* -> *Webhooks* and click *Add webhook*.
+.. Set the *Payload URL* to the Pipelines as Code controller public URL.
+.. Select the content type as *application/json*.
+.. Add a webhook secret and note it in an alternate location. With `openssl` installed on your local machine, generate a random secret.
```
Comment (rolfedh) [inline modules/odc-importing-codebase-from-git-to-create-application.adoc:75]: Neutral question: Would it be clearer to have these as two separate steps or together as they are?

**Review by rolfedh — APPROVED** [review]
Looks good to me! 
And a very minor question.

File: modules/odc-importing-codebase-from-git-to-create-application.adoc (hunk 4 above)
Comment (shreyasiddhartha) [reply modules/odc-importing-codebase-from-git-to-create-application.adoc:75]: These steps have been taken from another section mentioned in the Pipelines section. I'm guessing we can keep this in the same step as the purpose of this step is to add/generate a secret.

**Pull request merged**